crates/red_knot_project/src/lib.rs

```diff
@@ -128,6 +150,23 @@ impl Project {
         Arc::into_inner(result).unwrap().into_inner().unwrap()
     }
 
+    pub(crate) fn check_file(self, db: &dyn Db, file: File) -> Vec<Box<dyn Diagnostic>> {
```

Comment:
I included the rule diagnostics in `check_file` because an API consumer (e.g., the LSP) can decide only to call `check_file` but never `project.check` and would never see the diagnostics. 

The counter-argument is that these diagnostics aren't related to `file` so they shouldn't be shown. 

I'm not sure what the right answer is, but I defaulted to including them for now.